src/WebJobs.Script.WebHost/Diagnostics/SystemLogger.cs

```diff
@@ -37,6 +47,20 @@ public class SystemLogger : ILogger
             _debugStateProvider = debugStateProvider;
             _eventManager = eventManager;
             _scopeProvider = scopeProvider;
+            _logItemQueue = new List<LogItem>();
+            _logFlushTimer = new Timer(TimerFlush, null, 5000, 5000);
+
+            InitializeApplicationInfo();
+
+            if (standbyOptions.CurrentValue.InStandbyMode)
+            {
+                standbyOptions.OnChange(o =>
+                {
+                    // we're caching some information that changes when specialization occurs,
+                    // so we need to reinitialize
+                    InitializeApplicationInfo();
```

Comment:
In the PerfView analysis, repeated Environment calls were showing up, since we were writing tons of logs on every request, and as you can see below, for each log, we do a ton of work. Caching this app level info up front.